homeassistant/components/ecoforest/__init__.py

```diff
@@ -0,0 +1,39 @@
+"""The Ecoforest integration."""
+from __future__ import annotations
+
+import httpx
+from pyecoforest.api import EcoforestApi
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME, Platform
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+from .coordinator import EcoforestCoordinator
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Ecoforest from a config entry."""
+
+    host = entry.data[CONF_HOST]
+    auth = httpx.BasicAuth(entry.data[CONF_USERNAME], entry.data[CONF_PASSWORD])
+    api = EcoforestApi(host, auth)
```

Comment:
Can we do a validation here like in the config flow? Just to check if we can receive data?

Reply:
Isn't that covered by the fact we call `await coordinator.async_config_entry_first_refresh()`? I am still getting into the core internals, I have followed `ipma` and `enphase_envoy` integration as example for the structure.
 
I can also make the same call as in input validation of config flow, but in this case what is the way to handle errors?